docs/cFE Application Developers Guide.md

```diff
@@ -1303,14 +1303,13 @@ TypeCRC identifies which of the standard CRC polynomials to be used.  Currently,
 there are the following types available:
 
 ```
-CFE_MISSION_ES_CRC_8 – an 8-bit additive checksum calculation that returns a 32-bit value
-CFE_MISSION_ES_CRC_16 – a 16-bit additive checksum calculation that returns a 32-bit value
-CFE_MISSION_ES_CRC_32 – a 32-bit additive checksum calculation that returns a 32-bit value
-CFE_MISSION_ES_DEFAULT_CRC – the mission specified default CRC calculation
+CFE_ES_CRC_8 – an 8-bit additive checksum calculation that returns a 32-bit value
+CFE_ES_CRC_16 – a 16-bit additive checksum calculation that returns a 32-bit value
+CFE_ES_CRC_32 – a 32-bit additive checksum calculation that returns a 32-bit value
 ```
 
-Unless there is a specific interface with a specified CRC calculation,
-Applications must use the CFE_MISSION_ES_DEFAULT_CRC type.
```

Comment:
Given that using/referring to the `CFE_MISSION_ES_DEFAULT_CRC` symbol has been the documented way to call this function, I would not want to change this.  We should continue to define this symbol as it is likely used by apps for this described purpose.

Reply:
No worries.
By the way - I don't take it as pedantic at all. I'm a firm believer in doing things right the first time, and doing them thoroughly (if you're going to do it at all).
Thanks for the additional context.

Re. this change - is it the right approach to leave the `CFE_MISSION_ES_DEFAULT_CRC` #defined and simply setting it to the value of `CFE_ES_CrcType_CRC_16`?
If not, can you advise what would be the best approach?
Cheers

Reply:
Yes, because the doc said to use `CFE_MISSION_ES_DEFAULT_CRC` and we can assume most/all callers are doing it this way, just by keeping it defined but changing it to `CFE_ES_CrcType_CRC_16` should be relatively safe and low risk of breakage.  There is always the possibility that someone referenced `CFE_MISSION_ES_CRC_16` directly, but that is unlikely enough that I'm not too worried about it.

I do think that the extra option is essentially "dead code" because nobody ever implemented anything else, and thus could be justified in removal, but because it's part of the public ES API we would have to go through the deprecation process (i.e. make a new API and but the old behind an OMIT_DEPRECATED switch, keeping both calls for at least one release cycle, letting users test and fix their app code, before removing the old one outright).  Try to avoid hard cuts/step changes in the API unless absolutely necessary.